parquet-format-structures/src/main/java/org/apache/parquet/format/Util.java

```diff
@@ -155,13 +226,33 @@ public void addRowGroup(RowGroup rowGroup) {
     public void addKeyValueMetaData(KeyValue kv) {
       md.addToKey_value_metadata(kv);
     }
+    
+    @Override
+    public void setEncryptionAlgorithm(EncryptionAlgorithm encryptionAlgorithm) {
+      md.setEncryption_algorithm(encryptionAlgorithm);
+    }
+
+    @Override
+    public void setFooterSigningKeyMetadata(byte[] footerSigningKeyMetadata) {
+      md.setFooter_signing_key_metadata(footerSigningKeyMetadata);
+    }
   }
 
   public static void readFileMetaData(InputStream from, FileMetaDataConsumer consumer) throws IOException {
-    readFileMetaData(from, consumer, false);
+    readFileMetaData(from, consumer, (BlockCipher.Decryptor) null, (byte[]) null);
   }
-
+  
+  public static void readFileMetaData(InputStream from, FileMetaDataConsumer consumer, 
+      BlockCipher.Decryptor decryptor, byte[] AAD) throws IOException {
+    readFileMetaData(from, consumer, false, decryptor, AAD);
+  }
+  
   public static void readFileMetaData(InputStream from, final FileMetaDataConsumer consumer, boolean skipRowGroups) throws IOException {
+    readFileMetaData(from, consumer, skipRowGroups, (BlockCipher.Decryptor) null, (byte[]) null);
```

Comment:
I would omit the casting since it doesn't add any value.

Reply:
The value is in compiler detection of buggy calls to wrong methods (wrong signature), causes by typeless nulls.

Reply:
caused

Reply:
Could you please paste what kind of compiling error you met?  Maybe we can set an option in the compiler?

Reply:
I removed the extra casts, and didn't get any compile error. I'm also curious what was the error message you got.

Reply:
Yep, it compiles, but the reason is to prevent compiler to cast this to an overloaded method with a different object type in place of the decryptor etc. A sort of precaution.

Reply:
I can't imagine any scenario where this could happen. I think if there's an overloaded method with different parameters, and the compile can't figure out which method to call, then it would be a compilation error. In that case casting is the way to give hint to the compiler, that you'd like to call the method with BlockCipher.Decryptor and byte[] parameters, but in this case the method invocation is clear without casting too.

Reply:
I see. Ok, looks like we have two options: 1) leave the code as is - the cast helps code readers to understand what is nulled (in this case, that we call methods without encryption); also, it's  harmless 2) remove the casts in this and other pull requests. 
Please let me know your preference, I'm fine with either option.

Reply:
I personally prefer more compact code, and looks like @Fokko thinks the same, so if you don't mind, then let's remove those casts, that are not necessary and don't add any value!

Reply:
Ok, will do that.